src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -90,43 +81,35 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
                                  aggregates: Seq[AggregateExpression],
                                  source: TiDBRelation,
                                  selectRequest: TiSelectRequest = new TiSelectRequest): TiSelectRequest = {
-    aggregates.foreach(expr =>
-      expr match {
-        case aggExpr: AggregateExpression =>
-          aggExpr.aggregateFunction match {
-            case Average(_) =>
-              assert(false, "Should never be here")
-            case Sum(BasicExpression(arg)) => {
-              selectRequest.addAggregate(new TiSum(arg),
-                fromSparkType(aggExpr.aggregateFunction.dataType))
-            }
-            case Count(args) => {
-              val tiArgs = args.flatMap(BasicExpression.convertToTiExpr)
-              selectRequest.addAggregate(new TiCount(tiArgs: _*),
-                fromSparkType(aggExpr.aggregateFunction.dataType))
-            }
-            case Min(BasicExpression(arg)) =>
-              selectRequest.addAggregate(new TiMin(arg),
-                fromSparkType(aggExpr.aggregateFunction.dataType))
-
-            case Max(BasicExpression(arg)) =>
-              selectRequest.addAggregate(new TiMax(arg),
-                fromSparkType(aggExpr.aggregateFunction.dataType))
-
-            case First(BasicExpression(arg), _) =>
-              selectRequest.addAggregate(new TiFirst(arg),
-                fromSparkType(aggExpr.aggregateFunction.dataType))
-
-            case _ => None
```

Comment:
Returning a `None` in a `foreach` closure call looks weird since the closure returns `Unit`. The original `None` is simply discarded.